### PR TITLE
[Fix: pixart-alpha] random 512px resolution bug

### PR DESCRIPTION
**Pull request by lawrence-cj**
# What does this PR do?

This PR fixes the PixArt-Alpha pipeline when setting `use_resolution_binning=True` in 512px images generateion. 
Specifically, we try to add a ASPECT_RATIO_512_BIN in [pipeline_pixart_alpha](https://github.com/huggingface/diffusers/blob/fix/pixart-alpha-multiple-images/src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py) for random 512px image generation, and choose the correct bin according to the sample size of Pipeline . Cc: @sayakpaul

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]
Looks nice! 

Could you also add a slow test? That would be similar to:
https://github.com/huggingface/diffusers/blob/ff573ae245166d8b958950f68eb65ea766f07f69/tests/pipelines/pixart/test_pixart.py#L395

**Comment by lawrence-cj** [comment]
## Comparison
|main|PR|
|:----|:----|
|  | |
|  |  |

Cc: @sayakpaul, I have tested on the local machine and will add a test file in this PR.

**Review by patrickvonplaten — APPROVED** [review]
Nice, good job!

File: tests/pipelines/pixart/test_pixart.py [hunk 1]
```diff
@@ -410,3 +410,22 @@ def test_pixart_1024_without_resolution_binning(self):
...
+        image = pipe(prompt, generator=generator, num_inference_steps=5, output_type="np").images
+        image_slice = image[0, -3:, -3:, -1]
+
+        generator = torch.manual_seed(0)
+        no_res_bin_image = pipe(
+            prompt, generator=generator, num_inference_steps=5, output_type="np", use_resolution_binning=False
+        ).images
+        no_res_bin_image_slice = no_res_bin_image[0, -3:, -3:, -1]
+
+        assert not np.allclose(image_slice, no_res_bin_image_slice, atol=1e-4, rtol=1e-4)
```
Comment (yiyixuxu) [inline tests/pipelines/pixart/test_pixart.py:431]: I don't understand this test 

I think `use_resolution_binning` only come into effect when you want to generate non square images, i.e. when `height` and `width` is not the same - here both are `None` so it should not make a difference

why do we expect the output for `use_resolution_binning = True` and `use_resolution_binning=False` to be different?

Comment (sayakpaul) [reply tests/pipelines/pixart/test_pixart.py:431]: So, it's basically testing that here, actually i.e., when resolution binning is not effective when the height == width. But I guess it's not comprehensive enough that way. I will modify it accordingly.

**Comment by sayakpaul** [comment]
Addressed https://github.com/huggingface/diffusers/pull/5842#discussion_r1399891533. Will merged once the CI is green.

**Pull request merged**